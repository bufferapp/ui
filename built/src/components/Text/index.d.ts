export { default } from './Text';
//# sourceMappingURL=index.d.ts.map