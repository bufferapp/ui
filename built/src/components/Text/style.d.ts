export declare const Span: any;
export declare const Paragraph: any;
export declare const H1: any;
export declare const H2: any;
export declare const H3: any;
export declare const Label: any;
export declare const Help: any;
//# sourceMappingURL=style.d.ts.map