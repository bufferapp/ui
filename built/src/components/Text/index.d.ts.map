{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["../../../../src/components/Text/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,QAAQ,CAAA"}