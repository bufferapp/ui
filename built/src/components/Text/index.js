export { default } from './Text';
//# sourceMappingURL=index.js.map