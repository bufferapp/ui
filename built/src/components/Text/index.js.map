{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../../src/components/Text/index.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,OAAO,EAAE,MAAM,QAAQ,CAAA","sourcesContent":["export { default } from './Text'\n"]}