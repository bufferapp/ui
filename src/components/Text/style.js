import styled from 'styled-components'
import textColors, { redDark, grayDarker } from '../style/colors'
import {
  fontFamily,
  fontSize,
  fontWeight,
  fontWeightBold,
  fontWeightMedium,
  lineHeight,
} from '../style/fonts'

export const Span = styled.span`
  color: inherit;
  font-family: ${fontFamily};
  font-size: inherit;
  font-weight: inherit;
  line-height: inherit;
`

export const Paragraph = styled.p`
  color: ${(props) => textColors[props.color]};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${fontWeight};
  line-height: 20px;
`

export const H1 = styled.h1`
  color: ${(props) => textColors[props.color]};
  font-family: ${fontFamily};
  font-size: 32px;
  font-weight: ${fontWeightBold};
  line-height: 44px;
`

export const H2 = styled.h2`
  color: ${(props) => textColors[props.color]};
  font-family: ${fontFamily};
  font-size: 24px;
  font-weight: ${fontWeightBold};
  line-height: 34px;
`

export const H3 = styled.h3`
  color: ${(props) => textColors[props.color]};
  font-family: ${fontFamily};
  font-size: 18px;
  font-weight: ${fontWeightBold};
  line-height: 28px;
`

export const Label = styled.label`
  color: ${(props) => textColors[props.color]};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${fontWeightMedium};
  line-height: ${lineHeight};
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  cursor: inherit;
`

export const Help = styled.label`
  color: ${({ hasError }) => (hasError ? redDark : grayDarker)};
  font-family: ${fontFamily};
  font-size: 14px;
  font-weight: ${fontWeightMedium};
  line-height: ${lineHeight};
`
