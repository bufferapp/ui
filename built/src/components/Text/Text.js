var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from 'react';
import PropTypes from 'prop-types';
import * as Styles from './style';
const Text = (_a) => {
    var { children, type } = _a, props = __rest(_a, ["children", "type"]);
    switch (type) {
        case 'h1':
            return React.createElement(Styles.H1, Object.assign({}, props), children);
        case 'h2':
            return React.createElement(Styles.H2, Object.assign({}, props), children);
        case 'h3':
            return React.createElement(Styles.H3, Object.assign({}, props), children);
        case 'p':
            return React.createElement(Styles.Paragraph, Object.assign({}, props), children);
        case 'label':
            return React.createElement(Styles.Label, Object.assign({}, props), children);
        case 'help':
            return React.createElement(Styles.Help, Object.assign({}, props), children);
        default:
            return React.createElement(Styles.Span, null, children);
    }
};
Text.propTypes = {
    children: PropTypes.node,
    hasError: PropTypes.bool,
    htmlFor: PropTypes.string,
    color: PropTypes.string,
    type: PropTypes.string,
    as: PropTypes.string,
};
Text.defaultProps = {
    children: undefined,
    hasError: false,
    htmlFor: undefined,
    color: 'grayDarker',
    type: 'span',
};
export default Text;
//# sourceMappingURL=Text.js.map