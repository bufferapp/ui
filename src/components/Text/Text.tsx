import React from 'react'

import PropTypes from 'prop-types'
import * as Styles from './style'

// @ts-expect-error TS(7031) FIXME: Binding element 'children' implicitly has an 'any'... Remove this comment to see the full error message
const Text = ({ children, type, ...props }) => {
  switch (type) {
    case 'h1':
      return <Styles.H1 {...props}>{children}</Styles.H1>
    case 'h2':
      return <Styles.H2 {...props}>{children}</Styles.H2>
    case 'h3':
      return <Styles.H3 {...props}>{children}</Styles.H3>
    case 'p':
      return <Styles.Paragraph {...props}>{children}</Styles.Paragraph>
    case 'label':
      return <Styles.Label {...props}>{children}</Styles.Label>
    case 'help':
      return <Styles.Help {...props}>{children}</Styles.Help>
    default:
      return <Styles.Span>{children}</Styles.Span>
  }
}

Text.propTypes = {
  /** The actual text content. */
  children: PropTypes.node,
  /** It changes the color of the text to red. <br><i>This is only used for `help`</i>. */
  hasError: PropTypes.bool,
  /** It's the name of the input it refers to. <br><i>This is only used for `label`.</i> */
  htmlFor: PropTypes.string,
  /** The color can be: `white`, `gray`, `grayDark`, `blue` and `red`.<br><i>Not used for `span` and `help`.</i> */
  color: PropTypes.string,
  /** The type can be: `h1`, `h2`, `h3`, `p`, `label`, `help`. <br><i>If omitted will return a `span`</i> */
  type: PropTypes.string,
  /** If you want to keep all the styling you've applied to a component but just switch out what's being ultimately rendered (be it a different HTML tag or a different custom component), you can use the "as" prop to do this at runtime. */
  as: PropTypes.string,
}

Text.defaultProps = {
  children: undefined,
  hasError: false,
  htmlFor: undefined,
  color: 'grayDarker',
  type: 'span',
}

export default Text
