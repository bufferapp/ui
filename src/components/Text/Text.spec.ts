/* eslint-disable react/jsx-filename-extension */
// @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module 'jest... Remove this comment to see the full error message
import snap from 'jest-auto-snapshots'
import 'jest-styled-components'

import Text from './Text'

snap(Text, './Text.jsx')
