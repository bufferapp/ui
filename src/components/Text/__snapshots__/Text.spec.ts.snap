// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > Text Matches snapshot when boolean prop "hasError" is set to: "false" 1`] = `
<styled.span>
  <NodeFixture />
</styled.span>
`;

exports[`jest-auto-snapshots > Text Matches snapshot when passed all props 1`] = `
<styled.span>
  <NodeFixture />
</styled.span>
`;
